demo/login.cgi

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/python
+r"""
+Taken from https://github.com/plone/plone.session/blob/master/plone/session/tktauth.py
+It needed some modifications to work properly:
+- The IP must be encoded as a string (plone was using 4-bytes). The reason mod_authn_tkt
+does this is because an IP address might be IPv4 or it might be IPv6. An IPv6 address
+does not fit into 4 packed bytes (which is the size of an IPv4 address)
+- The timestamp must be encoded in little-endian (plone was using network/big-endian).
```

Comment:
That might be a bug.  mod_authn_tkt converts timestamp to big-endian with `htonl()` before hashing the timestamp.  However, it also uses `htonl()` in `refresh_cookie()`, which would undo the big-endian to little-endian on little-endian systems.  I'll review the code later today, but this looks like a bug.  For consistency, it should be big-endian throughout.  Thanks for finding this!

Reply:
With the following patch, the timestamps should be big-endian, including for ticket refresh.  (I have not yet pushed this patch)
```
--- a/mod_authn_tkt.c
+++ b/mod_authn_tkt.c
@@ -246,6 +246,7 @@ static int parse_ticket(authn_tkt *parsed)/*{{{*/
                                       sizeof(parsed->timestamp))) {
         return 0; /*(invalid hex encoding in timestamp)*/
     }
+    parsed->timestamp = htonl(parsed->timestamp);
 
     buffer_copy_string_len(&parsed->uid,
                            tkt + digest_hexlen + TIMESTAMP_HEXLEN,
```

Reply:
Oh, then the C implementation must be changed as well. Glad to be helpful anyway.
Sligthly off-topic; since it's my first Pull-Request in github, how do I do this after you push the bugfix to master? Rebase on original repo and then (force) push my fork? Or merge your repo and then commit the changes?

Reply:
If you force push to your branch, it will update this pull request.  I have not yet merged any changes.